examples/todo/src/controllers/model-admin.controller.ts

```diff
@@ -0,0 +1,242 @@
+           idInjection: false,
+           mysql: {schema: 'test', table: 'CoffeeShop'},
+         },
+         properties: {
+           id: {
+             type: 'Number',
+             required: true,
+             length: null,
+             precision: 10,
+             scale: 0,
+             id: 1,
+             mysql: {
+               columnName: 'id',
+               dataType: 'int',
+               dataLength: null,
+               dataPrecision: 10,
+               dataScale: 0,
+               nullable: 'N',
+             },
+           },
+           name: {
+             type: 'String',
+             required: false,
+             length: 512,
+             precision: null,
+             scale: null,
+             mysql: {
+               columnName: 'name',
+               dataType: 'varchar',
+               dataLength: 512,
+               dataPrecision: null,
+               dataScale: null,
+               nullable: 'Y',
+             },
+           },
+           city: {
+             type: 'String',
+             required: false,
+             length: 512,
+             precision: null,
+             scale: null,
+             mysql: {
+               columnName: 'city',
+               dataType: 'varchar',
+               dataLength: 512,
+               dataPrecision: null,
+               dataScale: null,
+               nullable: 'Y',
+             },
+           },
+       }}}
+     */
+  const jugglerDef = discoveredModels[Object.keys(discoveredModels)[0]];
+
+  return new ModelDefinition({
+    name: jugglerDef.name,
+    // TODO: convert from juggler/LB3 style to LB4
+    // For example, we need to transform array-type definitions.
+    properties: jugglerDef.properties,
+    settings: jugglerDef.options,
+  });
+}
+
+// TODO: move this function to '@loopback/repository'
+function defineModelClass<T extends typeof Model>(
+  base: T /* Model or Entity */,
+  definition: ModelDefinition,
+): T {
+  const modelName = definition.name;
+  const defineNamedModelClass = new Function(
+    'BaseClass',
+    `return class ${modelName} extends BaseClass {}`,
+  );
+  const modelClass = defineNamedModelClass(base) as T;
+  assert.equal(modelClass.name, modelName);
+  modelClass.definition = definition;
+  console.log('Defined model class %s', definition.name);
+  return modelClass;
+}
+
+// This will be implemented by
+// https://github.com/strongloop/loopback-next/issues/2036
+// In particular https://github.com/strongloop/loopback-next/issues/3737
+function bootModelApi(
+  app: TodoListApplication,
+  dsName: string,
+  modelClass: typeof Entity,
+  options: CrudRestControllerOptions,
+) {
+  const RepositoryClass = defineCrudRepositoryClass(modelClass);
+  inject(`datasources.${dsName}`)(RepositoryClass, undefined, 0);
+  const repoBinding = app.repository(RepositoryClass);
+  console.log('Defined repository %s', RepositoryClass.name);
+
+  const ControllerClass = defineCrudRestController(modelClass, options);
```

Comment:
Cross-posting from a Slack chat:

> In LB3, we have a feature to expose the models over REST or not via `public:true/false` parameter and we need this feature as well as we are not exposing the models via REST. We are accessing them in nodejs and don't want to expose those routes

Good point! This is certainly possible. If you look at how `bootModelApi` is implemented, it does two things: define a repository class to connect the model with a datasource, and then define a controller class to build REST API. You can skip controller creation to get a model that's accessible inside the server code only.